src/format.ts

```diff
@@ -0,0 +1,48 @@
+import BikramSambat from 'BikramSambat'
+import { NepaliMonthsData, type Format } from 'data'
+import { generateDateFormatOrder } from 'utils/generateDateFormatOrder'
+
+const MonthNames = NepaliMonthsData.map((month) => month.en)
+
+/**
+ * Formats a BikramSambat date into the specified format.
+ *
+ * @param {BikramSambat} date - The BikramSambat date to be formatted.
+ * @param {Format} dateFormat - The desired format for formatting the date.
+ * @returns {string} - The formatted date string.
+ * @example <caption>Format a BikramSambat date into `YYYY-MM-DD` format.</caption>
+ * const date = new BikramSambat(2077, 1, 1)
```

Comment:
```suggestion
 * const date = new BikramSambat(2077, 1, 1)
```
do we support this type of constructor? I don't think we do yet right?